test/geopackage/WKB_Test.cpp

```diff
@@ -0,0 +1,262 @@
+            0x02, 0x00, 0x00, 0x00,
+            0x03, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x34, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x34, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x01,
+            0x02, 0x00, 0x00, 0x00,
+            0x04, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3e, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3e, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x34, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3e, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40
+        };
+
+        // POLYGON ((0 0, 5 0, 5 5, 0 5, 0 0), (2 2, 3 2, 3 3, 2 3, 2 2))
+        this->wkb_bytes["polygon_with_holes"] = {
+            0x01,
+            0x03, 0x00, 0x00, 0x00,
+            0x02, 0x00, 0x00, 0x00,
+            0x05, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x05, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40, 
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40
+        };
+    }
+
+    void TearDown() override {}
+    
+    std::map<std::string, std::vector<uint8_t>> wkb_bytes;
+};
+
+TEST_F(WKB_Test, wkb_point_test) // also tests endianness
+{  
+    const auto geom_big    = boost::get<wkb::point_t>(wkb::read(this->wkb_bytes["point_big"]));
+    const auto geom_little = boost::get<wkb::point_t>(wkb::read(this->wkb_bytes["point"]));
+    EXPECT_NEAR(geom_big.get<0>(), geom_little.get<0>(), 0.000001);
```

Comment:
Stylistic thing: it's easier to read tolerances specified in scientific notation, so as not to have to count repeated `0`s

Reply:
I didn't know that C++ supported scientific notation, thanks! Implemented this in 5419798.